lib/P6doc.pm6

```diff
@@ -1,187 +1,356 @@
+            $final-docee = %prefixes{ $<choice>.lc };
+            last;
+        } else {
+            say "$word doesn't seem to apply here.\n";
+            next;
+        }
+    }
 
-	return $final-docee;
+    return $final-docee;
 }
 
 sub locate-curli-module($module) {
-	my $cu = try $*REPO.need(CompUnit::DependencySpecification.new(:short-name($module)));
-	unless $cu.DEFINITE {
-		fail "No such type '$module'";
-		#exit 1;
-	}
-	return ~ $cu.repo.prefix.child('sources/' ~ $cu.repo-id);
+    my $cu = try $*REPO.need(CompUnit::DependencySpecification.new(:short-name($module)));
+    unless $cu.DEFINITE {
+        fail "No such type '$module'";
+        #exit 1;
+    }
+    return ~ $cu.repo.prefix.child('sources/' ~ $cu.repo-id);
 }
 
 # see: Zef::Client.list-installed()
 # Eventually replace with CURI.installed()
 # https://github.com/rakudo/rakudo/blob/8d0fa6616bab6436eab870b512056afdf5880e08/src/core/CompUnit/Repository/Installable.pm#L21
 sub list-installed() is export {
-	my @curs       = $*REPO.repo-chain.grep(*.?prefix.?e);
-	my @repo-dirs  = @curs>>.prefix;
-	my @dist-dirs  = |@repo-dirs.map(*.child('dist')).grep(*.e);
-	my @dist-files = |@dist-dirs.map(*.IO.dir.grep(*.IO.f).Slip);
-
-	my $dists := gather for @dist-files -> $file {
-		if try { Distribution.new( |%(from-json($file.IO.slurp)) ) } -> $dist {
-			my $cur = @curs.first: {.prefix eq $file.parent.parent}
-			my $dist-with-prefix = $dist but role :: { has $.repo-prefix = $cur.prefix };
-			take $dist-with-prefix;
-		}
-	}
+    my @curs       = $*REPO.repo-chain.grep(*.?prefix.?e);
+    my @repo-dirs  = @curs>>.prefix;
+    my @dist-dirs  = |@repo-dirs.map(*.child('dist')).grep(*.e);
+    my @dist-files = |@dist-dirs.map(*.IO.dir.grep(*.IO.f).Slip);
+
+    my $dists := gather for @dist-files -> $file {
+        if try { Distribution.new( |%(from-json($file.IO.slurp)) ) } -> $dist {
+            my $cur = @curs.first: {.prefix eq $file.parent.parent}
+            my $dist-with-prefix = $dist but role :: { has $.repo-prefix = $cur.prefix };
+            take $dist-with-prefix;
+        }
+    }
+}
+
+###
+### NEXT
+###
+
+#| Create a Perl6::Documentable::Registry for the given directory
+sub compose-registry(
+    $topdir,
+    @dirs = ['Type'] ) returns Perl6::Documentable::Registry
+{
+    my $registry = process-pod-collection(
+        cache => False,
+        verbose => False,
+        topdir => $topdir,
+        dirs => @dirs
+    );
+    $registry.compose;
+
+    return $registry
+}
+
+#| Receive a list of paths to pod files and process them, return a list of
+#| Perl6::Documentable objects
+sub process-type-pods(IO::Path @files) returns Array[Perl6::Documentable] is export {
+    my Perl6::Documentable @results;
+
+    for @files.list -> $f {
+        my $documentable = process-pod-source(
+            kind => "type",
+            pod => load($f)[0],
+            filename => $f.Str,
+        );
+        @results.push($documentable);
+    }
+
+    return @results;
```

Comment:
You can leave off the word `return` here, and use just `@results`. I'm not sure if this is still a valid concern, but I was told (a recent while ago) that implicit returns are faster than explicit usage of `return`.